GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/UploadObjectOptions.cs

```diff
@@ -18,6 +19,30 @@ public class UploadObjectOptions
         /// </summary>
         public const int MinimumChunkSize = ResumableUpload<Data.Object>.MinimumChunkSize;
 
+        /// <summary>
+        /// Precondition for upload: if the object's generation does not match the
+        /// given generation, it is not uploaded.
+        /// </summary>
+        public long? IfGenerationMatch;
+
+        /// <summary>
+        /// Precondition for upload: if the object's generation matches the
+        /// given generation, it is not uploaded.
+        /// </summary>
+        public long? IfGenerationNotMatch;
+
+        /// <summary>
+        /// Precondition for upload: if the object's meta-generation does not match the
```

Comment:
Have changed all the comments - let me know what you think. I'll amend the download ones in the same way if you like these.